Code/GraphMol/FileParsers/CMLWriter.cpp

```diff
@@ -0,0 +1,209 @@
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#include "FileParsers.h"
+#include <fstream>
+#include <sstream>
+#include <boost/format.hpp>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/xml_parser.hpp>
+#include <RDGeneral/Invariant.h>
+
+#if __cplusplus >= 201402L && __has_cpp_attribute(fallthrough)
+#define FALLTHROUGH [[fallthrough]]
+#else
+#define FALLTHROUGH
+#endif
+
+namespace RDKit {
+
+namespace {
+boost::property_tree::ptree molToPTree(const ROMol& mol, int confId,
+                                       bool kekulize) {
+  if (!mol.getNumConformers()) {
+    BOOST_LOG(rdErrorLog)
+        << "Cannot write molecules with no conformers to CML format\n";
+    return {};
+  }
+
+  auto romol{mol};
+  auto& rwmol = static_cast<RWMol&>(romol);
+  if (kekulize) MolOps::Kekulize(rwmol);
+  const auto& conf = rwmol.getConformer(confId);
+
+  boost::property_tree::ptree pt;
+  // prefix namespaces
+  auto& root = pt.add("cml:cml", "");
+  root.put("<xmlattr>.xmlns:cml", "http://www.xml-cml.org/schema");
+  root.put("<xmlattr>.xmlns:convention", "http://www.xml-cml.org/convention/");
+  root.put("<xmlattr>.convention", "convention:molecular");
+
+  auto& molecule = root.add("cml:molecule", "");
+
+  // molecule/@id MUST start with an alphabetical character
+  // http://www.xml-cml.org/convention/molecular#molecule-id
+  const auto molecule_id_default_prefix = "m";
+  molecule.put("<xmlattr>.id",
+               boost::format{"%1%%2%"} % molecule_id_default_prefix % confId);
+
+  std::string name;
+  rwmol.getPropIfPresent(common_properties::_Name, name);
+  if (!name.empty()) molecule.put("cml:name", name);
+
+  int mol_formal_charge = 0;
+  unsigned mol_num_radical_electrons = 0;
+
+  // atom/@id MUST start with an alphabetical character
+  // http://www.xml-cml.org/convention/molecular#atom-id
+  const auto atom_id_prefix = "a";
+
+  auto& atomArray = molecule.put("cml:atomArray", "");
+  for (unsigned i = 0, nAtoms = rwmol.getNumAtoms(); i < nAtoms; i++) {
+    const auto& pos = conf.getAtomPos(i);
+    auto& atom = atomArray.add("cml:atom", "");
+    const auto& a = rwmol.getAtomWithIdx(i);
+
+    atom.put("<xmlattr>.id", boost::format{"%1%%2%"} % atom_id_prefix % i);
+    if (a->getAtomicNum()) {
+      atom.put("<xmlattr>.elementType", a->getSymbol());
+    } else {
+      atom.put("<xmlattr>.elementType", "Du");  // dummy
+    }
+
+    const auto charge = a->getFormalCharge();
+    mol_formal_charge += charge;
+    atom.put("<xmlattr>.formalCharge", charge);
+
+    atom.put("<xmlattr>.hydrogenCount", a->getTotalNumHs());
+
+    const auto isotope = a->getIsotope();
+    if (isotope) atom.put("<xmlattr>.isotopeNumber", isotope);
+
+    const auto n_rad_es = a->getNumRadicalElectrons();
+    mol_num_radical_electrons += n_rad_es;
+    atom.put("<xmlattr>.spinMultiplicity", n_rad_es + 1u);
```

Comment:
This doesn't look right.

paired electrons = multiplicity = 1
one unpaired electron == spin multiplicity = 2
two unpaired electron == spin multiplicity can be  1 or 3 depending on opposite or same spine

Is this doing something different?

Reply:
You are correct.
There is no spin information, so `spinMultiplicity` attribute is added only in the case of zero or one radical electron.